### ISSUE TITLE
Allow upgrade to Faraday 0.12

### ISSUE THREAD

**Issue by stanhu** [lostisland/faraday_middleware#154]
Faraday 0.12.0.1 is out: https://github.com/lostisland/faraday/releases

It looks like Faraday Middleware depends on Faraday < 1.0: https://github.com/lostisland/faraday_middleware/blob/master/faraday_middleware.gemspec#L16

**Comment by pengwynn** [comment]
I think that Faraday release should have been 0.12.1 instead of 0.12.0.1. @iMacTia any reason for going four deep on that one?

**Comment by iMacTia** [comment]
0.12.0.1 should be < 1.0, which issues are you having exactly @stanhu ?
@pengwynn usually I don't use the 4th group on the version and the 3rd is normally for hotfixes. in this case the code on 0.12.0 and 0.12.0.1 is exactly the same, I just had the bump the version up to re-release on Rubygems, hence I opted for the 4th group.
I did the same with `faraday_middleware` when the same happened and it was perfectly fine.
